Marlin/SanityCheck.h

```diff
@@ -329,12 +329,14 @@
 #endif
 
 /**
- * A dual nozzle x-carriage with switching servo
+ * A Dual Nozzle carriage with switching servo
  */
 #if ENABLED(SWITCHING_NOZZLE)
-  #if ENABLED(SINGLENOZZLE)
+  #if ENABLED(DUAL_X_CARRIAGE)
+    #error "SWITCHING_NOZZLE and DUAL_X_CARRIAGE are incompatible."
```

Comment:
It was the nozzle switching part that was most incompatible, because nozzle offsets are used to define carriage offsets. But we can do switching extruders with dual X, no problem.